src/azure-cli/azure/cli/command_modules/hdinsight/_validators.py

```diff
@@ -48,9 +48,19 @@ def validate_subnet(cmd, namespace):
 
 # Validate managed identity.
 def validate_msi(cmd, namespace):
-    namespace.assign_identity = HDInsightValidator(
-        resource_type='Microsoft.ManagedIdentity/userAssignedIdentities',
-        resource_name=namespace.assign_identity).validate(cmd, namespace)
+    if namespace.assign_identity is None:
+        namespace.assign_identity = []
+    elif isinstance(namespace.assign_identity, str):
+        namespace.assign_identity = [namespace.assign_identity]
+
+    validated_identities = []
```

Comment:
better to refactor this method, the new change will overwrite the validate functionality because it can be empty.
However, its origin functionality is validating the provided msi is validate or not.

Reply:
1. This validator executes whether or not `--assign-identity` is added to the command.
2. If the value here is required not to be empty in the validator,  `assign-identity`  must a required field. But not actually a required field.
3. We can see that the validator will perform verification only if it judges 'self.resource_name'(value of assigned-identity), otherwise nothing will be done.
![image](https://github.com/user-attachments/assets/f4cb5e4a-7de3-4b21-8378-0ebff0fd1946)

Reply:
Resolved this.